app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -18,49 +18,81 @@ class GetUspsProofingResultsJob < ApplicationJob
 
   discard_on GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError
 
+  def enrollment_attributes(enrollment, enrollment_completed)
+    {
+      enrollment_code: enrollment.enrollment_code,
+      enrollment_id: enrollment.id,
+      minutes_since_last_status_check: enrollment.minutes_since_last_status_check,
+      minutes_since_last_status_update: enrollment.minutes_since_last_status_update,
+      minutes_to_completion: enrollment_completed ? enrollment.minutes_since_established : nil,
```

Comment:
I wonder if we could arrange it so that the enrollment is updated first, then infer this from the enrollment's new `status`.

```suggestion
      minutes_to_completion: COMPLETE_STATUSES.include?(enrollment.status) ? enrollment.minutes_since_established : nil,
```

On the other hand, it could be a bit too "magical" (not in a positive way) and prone to human error by relying on specific ordering.

Part of the thinking is that boolean positional arguments like `enrollment_completed` can be hard to grok when glancing at code like this, without taking the extra step to refer back to the method's argument signature:

```rb
enrollment_attributes(enrollment, true) # What is `true` referring to?
```

Reply:
Yeah, I tried something along these lines but it got hairy with the `status_last_updated_at` timestamp. I had another idea while responding to this comment though -- what do you think of this approach? 8e5667d65

Reply:
Ah, yeah, I hadn't considered that. I think I might have preferred it how it was previously. Maybe a compromise solution is to make the "completed" argument a keyword-argument, so that it's less susceptible to the mystery-boolean-argument problem?

i.e.

```rb
enrollment_attributes(enrollment, complete: true)
```

Reply:
Works for me :+1:

Reply:
Updated in c1053bfde